develop/automatic-testing.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Automatic Testing
+description: A guide to writing automatic tests with Fabric Loader JUnit.
+authors:
+  - kevinthegreat1
+---
+
+# Automatic Testing {#automatic-testing}
+
+This page explains how to write code to automatically test parts of your mod.
+There are two ways to automatically test your mod:
+units tests with Fabric Loader JUnit or game tests with the Gametest framework from Minecraft.
+Units tests should be used to test components of your code such as methods and helper classes
+while gametests spins up an entire Minecraft client and server to run your tests
+which makes it more suitable for testing features and gameplay.
+Currently, this guide only covers unit testing.
+
+## Unit Testing with Fabric Loader JUnit {#unit-testing-with-fabric-loader-junit}
+
+Due to the nature of Minecraft modding such as obfuscation, simply adding and using JUnit normally will not work.
+That's why Fabric provides Fabric Loader JUnit, a JUnit plugin that enables unit testing in Minecraft.
+
+### Setting up Fabric Loader JUnit {#setting-up-fabric-loader-junit}
+
+First, we need to add Fabric Loader JUnit to the development environment. Add the following to your dependencies in your `build.gradle`:
+
+@[code lang=groovy transcludeWith=:::automatic-testing:1](@/reference/build.gradle)
+
+Then, we need to tell gradle to use Fabric Loader JUnit for testing. You can do so by adding the following code the to your `build.gradle`:
+
+@[code lang=groovy transcludeWith=:::automatic-testing:2](@/reference/build.gradle)
+
+#### Split Sources {#split-sources}
+
+If you are using split sources, you also need to add either the client or server source set to the test source set.
```

Comment:
Nothing to do for now, but do note I plan to fix this for Loom 1.8: https://github.com/FabricMC/fabric-loom/issues/1060